tests/unit/synapseclient/unit_test_tables.py

```diff
@@ -119,6 +140,111 @@ def test_cast_values__list_type():
     ]
 
 
+def test_convert_df_date_cols_to_datetime_empty_df():
+    test_df = pd.DataFrame()
+    date_columns = []
+    test_df_output = _convert_df_date_cols_to_datetime(test_df, date_columns)
+    assert test_df_output.empty
+
+
+def test_convert_df_date_cols_to_datetime_invalid_epoch_time(sample_df_datetime):
+    date_columns = ["epoch_time1"]
+    with pytest.raises(ValueError):
+        test_df_output = _convert_df_date_cols_to_datetime(
+            sample_df_datetime, date_columns
+        )
+
+
+def test_convert_df_date_cols_to_datetime_empty_vals(sample_df_datetime):
+    date_columns = ["epoch_time2"]
+    test_df_output = _convert_df_date_cols_to_datetime(sample_df_datetime, date_columns)
+    test_df_epoch_time2 = test_df_output[["epoch_time2"]]
+    expected_date_df = pd.DataFrame(
+        {
+            "epoch_time2": [
+                "2005-02-01 05:00:00+00:00",
+                pd.NaT,
+                pd.NaT,
+                pd.NaT,
+                pd.NaT,
+            ],
+        }
+    ).astype("datetime64[ns, UTC]")
+    assert_frame_equal(test_df_epoch_time2, expected_date_df)
+
+
+def test_convert_df_date_cols_to_datetime_invalid_date_columns():
+    test_df = pd.DataFrame(
+        {
+            "epoch_time1": [
+                "1107234000000",
+                "1107320400000",
+                "1107406800000",
+                "1107493200000",
+                "1107579600000",
+            ]
+        }
+    )
+    date_columns = ["invalid date column"]
+    with pytest.raises(ValueError):
+        test_df_output = _convert_df_date_cols_to_datetime(test_df, date_columns)
+
+
+def test_convert_df_date_cols_to_datetime():
+    # construct test dataframe and get date columns
+    test_df = pd.DataFrame(
+        {
+            "row": [1, 2, 3, 4, 5],
+            "epoch_time1": [
+                "1107234000000",
+                "1107320400000",
+                "1107406800000",
+                "1107493200000",
+                "1107579600000",
+            ],
+            "epoch_time2": [
+                "1107234000000",
```

Comment:
Last thing, add some test cases here.

* Negative number.
* Decimal number (not sure what that will even look like for the expected - so this could be optional.

Reply:
Thanks for the feedbacks! I will add the test